src/get-or-create-render-context.js

```diff
@@ -92,9 +104,9 @@ const createRenderContext = function(locationUrl, jsPackages, pathToClientEntryP
 
     // Now we execute inside the sandbox context each script package.
     let cumulativePackageSize = 0;
-    jsPackages.forEach(({content}) => {
+    jsPackages.forEach(({content, url}) => {
         // pkg is [url path, contents]
-        context.run(content);
+        context.run(content, {filename: url});
```

Comment:
The important bit; we add the url of the package as the filename so that node knows where the code is.